src/cmd/troubleshoot/builder.go

```diff
@@ -95,49 +101,122 @@ func (builder CommandBuilder) buildRun() func(*cobra.Command, []string) error {
 
 		apiReader := k8scluster.GetAPIReader()
 
-		troubleshootCtx := &troubleshootContext{
+		troubleshootCtx := troubleshootContext{
 			context:       context.Background(),
 			apiReader:     apiReader,
 			httpClient:    &http.Client{},
 			namespaceName: namespaceFlagValue,
-			dynakubeName:  dynakubeFlagValue,
 		}
 
 		results := NewChecksResults()
+		_ = runChecks(results, &troubleshootCtx, getDynakubeUnrelatedChecks()) // ignore error to avoid polluting pretty logs
+		resetLogger()
+
+		dynakubes, err := getDynakubes(troubleshootCtx, dynakubeFlagValue)
+		if err != nil {
+			return nil
+		}
+
+		runChecksForAllDynakubes(results, getDynakubeSpecificChecks(results, getDynakubeUnrelatedChecks()), dynakubes, apiReader)
```

Comment:
you are using the `getDynakubeUnrelatedChecks ` as `globalPrerequisites` in the `runChecksForAllDynakubes`

shouldn't we handle the error of `runChecks(results, &troubleshootCtx, getDynakubeUnrelatedChecks())` ? 
if something is wrong with the `globalPrerequisites`, then it does not make much sense to try to run any other tests